isisd/isis_vty_common.c

```diff
@@ -498,6 +500,88 @@ DEFUN (no_circuit_topology,
 	return isis_circuit_mt_enabled_set(circuit, mtid, false);
 }
 
+DEFUN (isis_bfd,
+       isis_bfd_cmd,
+       PROTO_NAME " bfd",
+       PROTO_HELP
+       "Enable BFD support\n")
+{
+	struct isis_circuit *circuit = isis_circuit_lookup(vty);
+	if (!circuit)
+		return CMD_ERR_NO_MATCH;
+
+	if (circuit->bfd_info
+	    && CHECK_FLAG(circuit->bfd_info->flags, BFD_FLAG_PARAM_CFG)) {
+		return CMD_SUCCESS;
+	}
+
+	isis_bfd_circuit_param_set(circuit, BFD_DEF_MIN_RX,
+				   BFD_DEF_MIN_TX, BFD_DEF_DETECT_MULT, true);
+
+	return CMD_SUCCESS;
+}
+
+#if HAVE_BFDD > 0
```

Comment:
Clarification: this `ifdef` (e.g. `HAVE_BFDD`) is a selector for FRR `bfdd` and `ptm-bfd`. I'm not sure if it is necessary to have this command here since it was not here before. The other daemons have this `ifdef` for the BFD `DEFUN`s because I didn't want to break people's configuration migration from `ptm-bfd` to `bfdd`. The timers / detect-multiplier and other should be configured under the `bfdd` daemon configuration node.

@donaldsharp what do you think about this? Do you want to keep `ptm-bfd` around for a while longer?

Reply:
Good to know, I mostly implemented this for reasons of cargo cult. Would be good to know whether Cumulus will move to `bfdd` or keep `ptmd` around for that.